controller/src/beerocks/cli/beerocks_cli_bml.cpp

```diff
@@ -2021,28 +2032,40 @@ int cli_bml::get_dcs_scan_results(const std::string &radio_mac, uint32_t max_res
             std::cout << __func__ << "ERROR: results_count(" << results_count << ")"
                       << " > max_results_size(" << max_results_size << ")" << std::endl;
         } else if ((status == 0) && (results_count > 0)) {
-            //todo: replace the temporary debug print with actual prints of the new results structure
             for (size_t i = 0; i < results_count; i++) {
                 auto res = results[i];
                 std::cout << "result[" << i << "]:" << std::endl
                           << "  ssid=" << res.ap_SSID << std::endl
                           << "  bssid=" << res.ap_BSSID << std::endl
-                          << "  mode=" << res.ap_Mode << std::endl
-                          << "  channel=" << (int)res.ap_Channel << std::endl
-                          << "  signal_strength=" << (int)res.ap_SignalStrength << std::endl
-                          << "  security_mode_enabled=" << res.ap_SecurityModeEnabled << std::endl
-                          << "  encryption_mode=" << res.ap_EncryptionMode << std::endl
+                          << "  mode=" << int(res.ap_Mode) << std::endl
+                          << "  channel=" << int(res.ap_Channel) << std::endl
+                          << "  signal_strength=" << int(res.ap_SignalStrength) << std::endl
+                          << "  security_mode_enabled="
+                          << string_from_int_array(res.ap_SecurityModeEnabled,
+                                                   BML_CHANNEL_SCAN_ENUM_LIST_SIZE)
+                          << std::endl
+                          << "  encryption_mode="
+                          << string_from_int_array(res.ap_EncryptionMode,
```

Comment:
Same here - there is a single encryption mode set at a given time. I don't understand why it is an array.